src/main/java/nextstep/courses/domain/SessionStatus.java

```diff
@@ -0,0 +1,13 @@
+package nextstep.courses.domain;
+
+import nextstep.courses.CannotEnrollStatusException;
+
+public enum SessionStatus {
+    READY, RECRUITING, END;
+
+    public void enableEnrollment(SessionStatus sessionStatus) {
+        if (!sessionStatus.equals(RECRUITING)) {
+            throw new CannotEnrollStatusException(CannotEnrollStatusException.MESSAGE);
+        }
+    }
```

Comment:
메서드 자체가 예외만을 던지는 검증 메서드로하기보다는 행위 자체를 매개변수로 전달할수도 있을 것 같아요. 

```suggestion
    public void enableEnrollment(SessionStatus sessionStatus, Consumer<SessionStatus> afterAction) {
        if (!sessionStatus.equals(RECRUITING)) {
            afterAction.accept(this);
        }
    }
```

Reply:
함수 지향적인 부분인 것 같아요. 객체지향적인 부분은 현재 코드가 좋아보이는데 아래와 같이 작성하면 장점이 있나요? 저는 위의 코드를 채택한 이유는 아래와 같아요.

* 매개변수가 줄어든다.
* Consumer<T>를 이용한 방식은 SessionStatues내부 동작을 다른 객체가 알아야한다.